test/new/db/cmd/cmd_macros

```diff
@@ -130,7 +130,7 @@ CMDS=<<EOF
 f foo.one = 1
 f foo.bar = 2
 f foo.cow = 3
-(;?v $$)() @@ foo*
+(_;?v $$)() @@ foo*
```

Comment:
Let's always have a name for macros. Anyway, even when you do `(;?v $$)`, a macro with an empty name will be created. This also solves a problem in the parser, where there may be an ambiguity when evaluating `(;?v $$)`.

One interpreation is `(` and `?v $$)`, another is `(;?v $$)`, but the parser has no way to understand which one you want. By forcing a macro name we can solve this problem.

Reply:
why. anonymous macros are nice and those don't need to be recorded anywhere. can be reword your explanation? i cant understand the reasoning

Reply:
Try:
```
[0x00000000]> (;?v $$)()
0x0
[0x00000000]> (
0 ( ; ?v $$)
```
As you can see, the "anonymous" macro is recorded anyway. And with an empty name. But that's not entirely a problem.

The thing is that if you have a parser that look at one or two tokens ahead, you cannot easily parse `(;?v $$)` because it could be interpreted as `(` + `?v $$)` (remember that `;` is a separator of commands and `(` is a command on its own. I understand this is not a strong reason, but I think at the moment making the parser more complicated just to support "nameless" macros is not good. Having `_` as a name is just one char more to type and it ensures the macro has always a name.

Reply:
i don't see why unnamed macros are a problem because if the ( command should ignore the ; until the )  only as a command, not in the middle of the blue

Reply:
I have already explained why, let me try to rephrase.

`(` is a command on its own, right? (it should list all registered macros)
`(;` is just the command `(` plus the command delimiter `;`.
`(; ?e hello` means `list all registered macros and then print the string "hello"`.
`(; ?e hello; ?e hi; ?v $$; pd 10; af; pdf` means `list all registered macros, print hello, print hi, show offset value; print disassembly 10, analyze function, print disassembly function`.

Now, consider that a parser (for what I know), can choose which rule a given set of tokens belong to only by looking at a couple of tokens ahead, so when you see:

`(; ?e` it is:
`<command-name> <command-delimiter> <command name>`
Now, this can be reduced only to:
`<command> <command>`
`<macro-command>` indeed MUST have a `)` at the end, so you cannot know whether there is a `)` just by looking at few tokens ahead.

On the other hand, by forcing a macro name you would have:
`<command-name> <arg> <command>`
which would match the rule for `<macro-command>` so the parser can immediately know that what's following is a macro command (whether it will contain the ending `)` or not. If it does not contain the ending `)`, then it's a parsing error).

Reply:
IMHO, `(_` is as short as `(`, so LGTM to me.